README.md

```diff
@@ -31,37 +36,255 @@ npm install @uploadcare/upload-client --save
 
 ## Usage
 
+### High Level API
+
+To get access to High Level API you need to create 
+an instance of `UploadClient` and provide settings to him:
+
 ```javascript
 import UploadClient from '@uploadcare/upload-client'
 import {FileFromEnum} from '@uploadcare/upload-client'
 
 const client = new UploadClient({publicKey: 'YOUR_PUBLIC_KEY'})
+```
 
-client.api.request({path: 'info', query})
-  .then(response => console.log(response.data))
+After that you can upload file from binary data:
 
-client.api.info(uuid)
-  .then(data => console.log(data.is_image))
+```javascript
+const fileUpload = client.fileFrom(FileFromEnum.Object, fileData)
 
-const directUpload = client.api.base(fileData)
+fileUpload
+  .then(file => console.log(file.uuid))
+```
 
-directUpload
-  .then(data => console.log(data.file))
+Or from URL with file:
 
-directUpload.onProgress = (progressEvent) => console.log(progressEvent.loaded / progressEvent.total)
+```javascript
+const fileURL = 'https://example.com/file.jpg'
+const fileUpload = client.fileFrom(FileFromEnum.URL, fileURL)
 
-const fileUpload = client.fileFrom(FileFromEnum.Object, fileData)
+fileUpload
+  .then(file => console.log(file.uuid))
+```
+
+Or even from `uploaded` before to Uploadcare file:
+```javascript
+const fileUUID = 'edfdf045-34c0-4087-bbdd-e3834921f890'
+const fileUpload = client.fileFrom(FileFromEnum.Uploaded, fileUUID)
 
 fileUpload
   .then(file => console.log(file.uuid))
+```
 
+You can track uploading progress:
```

Comment:
It seems that we'd better split the usage section into a couple of separate ones: uploading a file; tracking progress; etc.

Reply:
@dayton1987 Please, suppose your variant